clang/include/clang/AST/PrettyPrinter.h

```diff
@@ -47,14 +48,16 @@ class PrintingCallbacks {
 /// This type is intended to be small and suitable for passing by value.
 /// It is very frequently copied.
 struct PrintingPolicy {
-  /// Create a default printing policy for the specified language.
-  PrintingPolicy(const LangOptions &LO)
+  /// Create a default printing policy for the specified language and optional
+  /// target.
+  PrintingPolicy(const LangOptions &LO, const TargetInfo *TI = nullptr)
```

Comment:
I'm not sure we should be including TargetInfo.h here. Since there only seems to be one call, can we just add single bool parameter since that is probably easier than messing with LangOptions?

Reply:
I don't like that approach. I personally think TargetInfo.h is perfectly fine as it's in Basic. However, given there's only one user of the two-argument constructor it can just set the field directly. Alternatively, if we move the call to `Context->InitBuiltinTypes(*Target);` _after_ the call to `Context->setPrintingPolicy(...);`, we don't even need to set the field directly in ASTUnit, only ASTContext. It feels slightly wrong to do them in that order though and exposing some of the internal details, so I'm leaning towards just setting the field directly in ASTUnit if you still don't like using TargetInfo.h.

Reply:
I'm not opposed against using TargetInfo. I just though since that header is also pulled in by ASTContext, almost everything ends up including TargetInfo. I guess we could add a out-of-line helper function to avoid the `#include`

However, it probably doesn't make a noticeable difference in compile time so I would say do whatever you prefer.

Reply:
ASTContext.h itself pulls in TargetInfo.h...

Reply:
Oh, no, it forward-declares it

Reply:
I'm perfectly happy to merge this as is. We can always remove the include later.